### PR TITLE
Add retry to submit_event in trigger to avoid deadlock

### PR DESCRIPTION
**Pull request by NickYadance**
related: #27000

### CODE REVIEW

File: airflow/models/trigger.py [hunk 1]
```diff
@@ -116,7 +116,7 @@ def clean_unused(cls, session=None):
         session.query(Trigger).filter(Trigger.id.in_(ids)).delete(synchronize_session=False)
 
     @classmethod
-    @provide_session
```
Comment (uranusjr) [inline airflow/models/trigger.py]: I think you still need this?

Comment (NickYadance) [reply airflow/models/trigger.py]: https://github.com/apache/airflow/blob/4797a0322ed4b73bc34d3967376479a42d9ba190/airflow/utils/retries.py#L49-L53

Comment (ephraimbuddy) [reply airflow/models/trigger.py:119]: I think the session arg should no longer be equal to None

Comment (ephraimbuddy) [reply airflow/models/trigger.py]: I think the session arg should no longer be equal to None

Comment (ephraimbuddy) [reply airflow/models/trigger.py]: I think the session arg should no longer be equal to None

Comment (NickYadance) [reply airflow/models/trigger.py]: > I think the session arg should no longer be equal to None

yea right, replaced with run_with_db_retries.

**Comment by NickYadance** [comment]
close this one as i misconfigured git

**Pull request closed without merging**